linter/linter.go

```diff
@@ -147,18 +151,19 @@ func Run() bool {
 	_, graphResults := gr.CheckSequences()
 	seqResults.Union(graphResults)
 	if seqResults.AnyError {
+		errorPrinted := false // whether we printed anything
 		for _, seq := range sequences {
 			header, err := fmtHeader(seq, allSpecs)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "%s", err)
 			}
-			linter.printCheckResult(header, seqResults.Results[seq])
+			errorPrinted = linter.printCheckResult(header, seqResults.Results[seq]) || errorPrinted
 		}
-		return false
-	}
-
-	// 5. No errors occurred: print warnings, then exit.
-	if seqResults.AnyWarning {
+		if errorPrinted {
+			return false
+		}
+	} else if seqResults.AnyWarning {
```

Comment:
I think this should be just if, not else if. Else, if there are errors but not for the specified `--sequences`, you won't print any warnings for those sequences even if they exist.